libs/platform/src/lib/components/form/radio-group/radio-group.component.ts

```diff
@@ -0,0 +1,219 @@
+import {
+    EventEmitter,
+    Component,
+    ContentChildren,
+    QueryList,
+    Input,
+    ChangeDetectorRef,
+    ChangeDetectionStrategy,
+    AfterViewInit,
+    Output,
+    Self,
+    Optional,
+    ViewChildren,
+} from '@angular/core';
+import { NgControl, NgForm } from '@angular/forms';
+import { stateType } from '@fundamental-ngx/core';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { RadioButtonComponent } from './radio/radio.component';
+import { SelectItem } from '../data-model';
+import { CollectionBaseInput } from '../collection-base.input';
+
+// Increasing integer for generating unique ids for radio components.
+let nextUniqueId = 0;
+
+@Component({
+    selector: 'fdp-radio-group',
+    templateUrl: './radio-group.component.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class GroupRadioButtonComponent extends CollectionBaseInput implements AfterViewInit {
+    /** uniqly generated, if value not provided for id */
+    @Input()
+    id: string = `radio-group-${nextUniqueId++}`;
+
+    /**
+     * values of radio buttons. type will be of SelectItem or String
+     */
+    @Input()
+    list: Array<SelectItem | string>;
```

Comment:
What I mean by this list `list: Array<SelectItem | string>;` in relation to `CollectionBaseInput ` so to move this `@Input` into this parent, so we can use the same for CheckboxGroup. 
Since its CollectionBaseInput, it will have this `ist: Array<SelectItem | string>;`.

So you could remove it from here..